src/analysis/processing/qgsalgorithmkeepnbiggestparts.cpp

```diff
@@ -0,0 +1,171 @@
+/***************************************************************************
+                         qgsalgorithmkeepnbiggestparts.cpp
+                         ---------------------
+    begin                : March 2018
+    copyright            : (C) 2018 by Nyall Dawson
```

Comment:
Back to the Future

Credit should be something like:
(C) 2014 by Victor Olaya (Initial Python Work)
(C) 2023 by Nyall Dawson (C++ Port)

We never write it in QGIS, but we could add by default this kind of line that we have in other projects :
(C) QGIS Team and contributors for various fixes, improvements, etc

Reply:
I'm rather -1 to the running history, just because it's extremely rare in qgis code and I feel it's unfair to selectively credit only such a small portion of the code. I'd rather just stick to the convention which has developed over time of "if you make a new file put your name on it, but check git for an accurate history". And I'd be totally +1 to replacing all the copyright statements in bulk with something generic like "the qgis team" for this same reason.